src/transaction.rs

```diff
@@ -70,6 +70,118 @@ impl Default for ResolveCondition {
     }
 }
 
+/// The gas price setting to use.
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum GasPrice {
+    /// The standard estimated gas price from the node, this is usually the
+    /// median gas price from the last few blocks. This is the default gas price
+    /// used by transactions.
+    Standard,
+    /// A factor of the estimated gas price from the node. `GasPrice::Standard`
+    /// is equivalent to `GasPrice::Factor(1.0)`.
+    Factor(f64),
+    /// Specify a specific gas price to use for the transaction. This will cause
+    /// the transaction `SendFuture` to not query the node for a gas price
+    /// estimation.
+    Value(U256),
+}
+
+impl GasPrice {
+    /// A low gas price. Using this may result in long confirmation times for
+    /// transactions, or the transactions not being mined at all.
+    pub fn low() -> Self {
+        GasPrice::Factor(0.8)
+    }
+
+    /// A high gas price that usually results in faster mining times.
+    /// transactions, or the transactions not being mined at all.
+    pub fn fast() -> Self {
```

Comment:
Is this the logic of slow, average, fast on metamask? I thought they did some more complex quartile analysis. This we would have to be able to add a factor to each of the options (low, average, fas) individually.

Reply:
No, I assume that metamask uses ethgasstation which actually analyses the latest blocks (this assumption comes from the fact that it has a little graph displaying expected mining times for gas prices). I just picked those factors for fast and slow based on the graph it displays which I assumed wouldn't change much.

We can integrate ethgas station to have better slow/fast guesses, but that would be a more involved.

Reply:
Would it make sense to use the gnosis gas station?

Mainnet: https://safe-relay.gnosis.io/api/v1/gas-station/
Rinkeby: https://safe-relay.rinkeby.gnosis.io/api/v1/gas-station/

Reply:
We could do that, but that would complicate things a bit. Maybe we can add this functionality as a follow up PR?